src/textmate/src/grammar.ts

```diff
@@ -0,0 +1,412 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as tm from "@azure-tools/tmlanguage-generator";
+import path from "path";
+import plist from "plist";
+
+export const grammarPath = path.resolve(__dirname, '../bicep.tmlanguage');
+
+type IncludeRule = tm.IncludeRule<BicepScope>;
+type BeginEndRule = tm.BeginEndRule<BicepScope>;
+type MatchRule = tm.MatchRule<BicepScope>;
+type Grammar = tm.Grammar<BicepScope>;
+
+type BicepScope =
+  | "comment.block.bicep"
+  | "comment.line.double-slash.bicep"
+  | "constant.character.escape.bicep"
+  | "constant.numeric.bicep"
+  | "constant.language.bicep"
+  | "entity.name.type.bicep"
+  | "entity.name.function.bicep"
+  | "keyword.control.declaration.bicep"
+  | "keyword.other.bicep"
+  | "string.quoted.single.bicep"
+  | "string.quoted.multi.bicep"
+  | "variable.name.bicep"
+  | "variable.other.readwrite.bicep"
+  | "variable.other.property.bicep"
+  | "meta.objectliteral.bicep"
+  | "meta.object.member.bicep"
+  | "punctuation.definition.template-expression.begin.bicep" 
+  | "punctuation.definition.template-expression.end.bicep";
+
+const bounded = (text: string) => `\\b${text}\\b`;
+const after = (regex: string) => `(?<=${regex})`;
+const notAfter = (regex: string) => `(?<!${regex})`;
+const before = (regex: string) => `(?=${regex})`;
+const notBefore = (regex: string) => `(?!${regex})`;
+
+const meta: typeof tm.meta = tm.meta;
+const identifierStart = "[_$[:alpha:]]";
```

Comment:
>alpha [](start = 30, length = 5)

I'll have to remember to update this when we add UAX31 (subset of unicode chars) support in identifiers.